pkg/allocation/converters/converter_test.go

```diff
@@ -147,13 +150,72 @@ func TestConvertAllocationRequestToGameServerAllocation(t *testing.T) {
 			in:   nil,
 			want: nil,
 		},
+		{
+			name: "accepts deprecated metapatch field",
+			in: &pb.AllocationRequest{
+				Namespace:                    "",
+				MultiClusterSetting:          &pb.MultiClusterSetting{},
+				RequiredGameServerSelector:   &pb.LabelSelector{},
+				PreferredGameServerSelectors: []*pb.LabelSelector{},
+				Scheduling:                   pb.AllocationRequest_Distributed,
+				MetaPatch: &pb.MetaPatch{
+					Labels: map[string]string{
+						"a": "b",
+					},
+				},
+			},
+			want: &allocationv1.GameServerAllocation{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "",
+				},
+				Spec: allocationv1.GameServerAllocationSpec{
+					MultiClusterSetting: allocationv1.MultiClusterSetting{
+						Enabled: false,
+					},
+					Scheduling: apis.Distributed,
+					MetaPatch: allocationv1.MetaPatch{
+						Labels: map[string]string{
+							"a": "b",
+						},
+					},
+				},
+			},
+			skipConvertFromGSA: true,
+		},
+		{
+			name: "Rejects both metadata and metapatch field",
+			in: &pb.AllocationRequest{
+				Namespace:                    "",
+				MultiClusterSetting:          &pb.MultiClusterSetting{},
+				RequiredGameServerSelector:   &pb.LabelSelector{},
+				PreferredGameServerSelectors: []*pb.LabelSelector{},
+				Scheduling:                   pb.AllocationRequest_Distributed,
+				Metadata: &pb.MetaPatch{
+					Labels: map[string]string{
+						"a": "b",
+					},
+				},
+				MetaPatch: &pb.MetaPatch{
+					Labels: map[string]string{
+						"c": "d",
+					},
+				},
+			},
+			want:               nil,
+			skipConvertFromGSA: true,
+			wantError:          cmpopts.AnyError,
+		},
 	}
 	for _, tc := range tests {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
 
-			out := ConvertAllocationRequestToGSA(tc.in)
+			out, err := ConvertAllocationRequestToGSA(tc.in)
+			if !cmp.Equal(err, tc.wantError, cmpopts.EquateErrors()) {
+				t.Errorf("\"%s\" error: got %v, want %v", tc.name, err, tc.wantError)
+			}
+
 			if !assert.Equal(t, tc.want, out) {
 				t.Errorf("mismatch with want after conversion: \"%s\"", tc.name)
```

Comment:
This line could also be cleaned up if you are so inclined while you are in here 😄 the inner message could be moved into the `assert.Equals(...)` statement.

https://pkg.go.dev/github.com/stretchr/testify/assert#Equal